### ISSUE TITLE
print.citoarchitecture shows NAs for 2d cnn

### ISSUE THREAD

**Issue by cmteuffel** [citoverse/cito#118]
Hi,

found a glitch in print.citoarchitecture with a one channel (grayscale) 2D cnn. While the fitting seems to work, printing the architecture creates NAs. Could be the code is interpreting the channel dimension as 3D cnn. Hence also the odd stride and padding values.

`trainingData <- array(dim = c(nrow(events), 1, dim(2d_spectrum)))`

Convolution|Input: 326**x1**x47x75
           |Output: 128x-3x45**xNA**
           |Kernel: 5x3 **(stride=1x1x1, padding=0x0x0, dilation=1x1x1)**
           |Bias: TRUE
           |Activation: relu
           |Dropout: rate=0.3

MaxPool    |Input: 128x-3x45xNA
           |Output: 128x-5x45xNA
           |Kernel: 3x1 (stride=1x1x1, padding=0x0x0, dilation=1x1x1)

Convolution|Input: 128x-5x45xNA
           |Output: 128x-9x43xNA
           |Kernel: 5x3 (stride=1x1x1, padding=0x0x0, dilation=1x1x1)
           |Bias: TRUE
           |Batch normalization
           |Activation: relu
           |Dropout: rate=0.3

Convolution|Input: 128x-9x43xNA
           |Output: 128x-13x41xNA
           |Kernel: 5x3 (stride=1x1x1, padding=0x0x0, dilation=1x1x1)
           |Bias: TRUE
           |Activation: relu
           |Dropout: rate=0.3

Convolution|Input: 128x-13x41xNA
           |Output: 128x-17x39xNA
           |Kernel: 5x3 (stride=1x1x1, padding=0x0x0, dilation=1x1x1)
           |Bias: TRUE
           |Activation: relu
           |Dropout: rate=0.3

Convolution|Input: 128x-17x39xNA
           |Output: 256x-21x37xNA
           |Kernel: 5x3 (stride=1x1x1, padding=0x0x0, dilation=1x1x1)
           |Bias: TRUE
           |Activation: relu
           |Dropout: rate=0.3

Convolution|Input: 256x-21x37xNA
           |Output: 256x-25x35xNA
           |Kernel: 5x3 (stride=1x1x1, padding=0x0x0, dilation=1x1x1)
           |Bias: TRUE
           |Activation: relu
           |Dropout: rate=0.3

Convolution|Input: 256x-25x35xNA
           |Output: 256x-29x33xNA
           |Kernel: 5x3 (stride=1x1x1, padding=0x0x0, dilation=1x1x1)
           |Bias: TRUE
           |Activation: relu
           |Dropout: rate=0.3

Convolution|Input: 256x-29x33xNA
           |Output: 256x-33x31xNA
           |Kernel: 5x3 (stride=1x1x1, padding=0x0x0, dilation=1x1x1)
           |Bias: TRUE
           |Activation: relu
           |Dropout: rate=0.3

Convolution|Input: 256x-33x31**xNA**
           |Output: 256x-37x29**xNA**
           |Kernel: 5x3 (stride=1x1x1, padding=0x0x0, dilation=1x1x1)
           |Bias: TRUE
           |Activation: relu
           |Dropout: rate=0.3

Linear     **|Input: NA**
           |Output: 1024
           |Bias: TRUE
           |Activation: relu

Linear     |Input: 1024
           |Output: 2
           |Bias: TRUE
           |Activation: Depends on loss

Despite the obviously erroneous layer information fitting the network still works without error:

Loss at epoch 1: training: 0.694, validation: 0.694, lr: 0.00001
Loss at epoch 2: training: 0.694, validation: 0.694, lr: 0.00001
